infra/mongo.js

```diff
@@ -9,32 +9,39 @@ class Mongo {
 
     /**
      * starts the connection to mongo
-     * @returns {Promise<void>}
      */
     async init(config) {
         const clientSettings = { ...config.mongo.options };
-
         const logger = require('cf-logs').Logger('codefresh:infra:mongo'); // eslint-disable-line
         this.logger = logger;
 
         const { uri } = config.mongo;
         const dbName = config.mongo.dbName || getDbNameFromUri(uri);
-        const client = await MongoClient.connect(uri, clientSettings);
+        const client = new MongoClient(uri, clientSettings);
+        logger.info(`Mongo db name ${dbName}`);
+
+        try {
+            await client.connect();
+            logger.info('Mongo driver connected');
+        } catch (error) {
+            logger.error('Error connecting to MongoDB:', error);
+            throw error;
+        }
+
         this.client = client;
-        this.db = client.db(dbName);
-        logger.info('Mongo driver connected');
+        this.db = this.client.db(dbName);
+        logger.info('Mongo db initialized');
     }
 
 
     /**
      * stops the connection to mongo
-     * @returns {Promise<void>}
      */
-    stop() {
+    async stop() {
```

Comment:
@kim-codefresh I just realized that this can be more straightforward; IDK why we're checking `this.db`.
```suggestion
    async stop() {
      await this.client?.close();
    }
```

Reply:
Actually no, conditional chaining (`?.` syntax) is a native JS feature, so we can safely replace as in my first comment

Reply:
yeah sorry the issue is that our linter not allowing it